### PR TITLE
Require labels on subparts and sections

### PR DESCRIPTION
**Pull request by willbarton**
This change requires the label on subparts and sections to not be blank.

It was previously possible to save a section or subpart in the Wagtail admin without a label. This would then cause 500 errors when the label, which was expected to exist, did not.

This fixes some issues we've seen in production.

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

**Comment by willbarton** [comment]
@Scotchester how's this look? 
image

**Comment by willbarton** [comment]
I can't reproduce the loss of `?subpart=` on the query string though.

File: cfgov/regulations3k/models/django.py [hunk 1]
```diff
@@ -148,7 +159,10 @@ class Meta:
 
 
 class Subpart(models.Model):
-    label = models.CharField(max_length=255, blank=True)
+    label = models.CharField(
+        max_length=255,
+        validators=[validate_label],
```
Comment (Scotchester) [inline cfgov/regulations3k/models/django.py:164]: How about we add some help text, while we're at it? Try and stave off showing any validation errors in the first place.

```suggestion
        validators=[validate_label],
        help_text='Labels always require at least 1 alphanumeric character, '
                  'then any number of alphanumeric characters and hyphens.',
```

(sorry, this'll require recreating the migration)

File: cfgov/regulations3k/models/django.py [hunk 2]
```diff
@@ -201,7 +215,10 @@ class Meta:
 
 
 class Section(models.Model):
-    label = models.CharField(max_length=255, blank=True)
+    label = models.CharField(
+        max_length=255,
+        validators=[validate_label],
```
Comment (Scotchester) [inline cfgov/regulations3k/models/django.py:220]:
```suggestion
        validators=[validate_label],
        help_text='Labels always require at least 1 alphanumeric character, '
                  'then any number of alphanumeric characters and hyphens.',
```

File: cfgov/regulations3k/models/django.py [hunk 3]
```diff
@@ -16,6 +17,16 @@
 import regdown
 
 
+# Labels always require at least 1 alphanumeric character, then any number of
+# alphanumeric characters and hyphens.
+label_re_str = r'[\w]+[-\w]*'
+validate_label = RegexValidator(
+    re.compile(r'^' + label_re_str + r'$'),
+    'Enter a valid “label” consisting of letters, numbers, and hyphens.',
```
[the remark this answers is not in the document]
Comment (willbarton) [reply cfgov/regulations3k/models/django.py:25]: I was taking [the Django slug validator](https://github.com/django/django/blob/master/django/core/validators.py#L228) as an example here:

```python
    _('Enter a valid “slug” consisting of letters, numbers, underscores or hyphens.'),
```

[the remark this answers is not in the document]
Comment (Scotchester) [reply cfgov/regulations3k/models/django.py:25]: Fair enough.

Comment (higs4281) [inline cfgov/regulations3k/models/django.py:25]: Might help to add "with no spaces"
Same with sections.

File: cfgov/regulations3k/models/django.py [hunk 4]
```diff
@@ -148,7 +159,12 @@ class Meta:
 
 
 class Subpart(models.Model):
-    label = models.CharField(max_length=255, blank=True)
+    label = models.CharField(
+        max_length=255,
+        validators=[validate_label],
+        help_text='Labels always require at least 1 alphanumeric character, '
+                  'then any number of alphanumeric characters and hyphens.',
```
Comment (higs4281) [inline cfgov/regulations3k/models/django.py:166]: Could we add "No spaces."?

Comment (willbarton) [reply cfgov/regulations3k/models/django.py]: Done.

File: cfgov/regulations3k/models/django.py (hunk 3 above)
Comment (willbarton) [reply cfgov/regulations3k/models/django.py:25]: Done.

Comment (higs4281) [reply cfgov/regulations3k/models/django.py:25]: Not seeing this change.

Comment (higs4281) [reply cfgov/regulations3k/models/django.py:25]: Perhaps:

```suggestion
        'Enter a valid “label” consisting of letters, numbers, hyphens, and no spaces.',  # noqa
```

Comment (willbarton) [reply cfgov/regulations3k/models/django.py:25]: Oh, sorry, I missed this one.

**Pull request merged**